sqlglot/dialects/doris.py

```diff
@@ -22,6 +22,28 @@ def _lag_lead_sql(self, expression: exp.Lag | exp.Lead) -> str:
     )
 
 
+# Accept both DATE_TRUNC(datetime, unit) and DATE_TRUNC(unit, datetime)
+def _build_date_trunc(args: list[exp.Expression]) -> exp.Expression:
+    a0, a1 = seq_get(args, 0), seq_get(args, 1)
+
+    def _is_unit_like(e: exp.Expression | None) -> bool:
+        if not (isinstance(e, exp.Literal) and e.is_string):
+            return False
+        text = e.this
+        return not any(ch.isdigit() for ch in text)
+
+    # Determine which argument is the unit
+    unit_first = _is_unit_like(a0) and not _is_unit_like(a1)
+    if unit_first:
+        unit, this = a0, a1
+    else:
+        this, unit = a0, a1
```

Comment:
I think we can simplify this a little?
```suggestion
    unit, this = a0, a1 if _is_unit_like(a0) else a1, a0
```